workspaces/adventure-pack/goodies/java/src/integer_digits/AP.java

```diff
@@ -0,0 +1,30 @@
+package integer_digits;
+
+import iterable_int_stream.IterableIntStream;
+import java.util.stream.IntStream;
+
+public final class AP {
+
+  /**
+   * Get the digits in the given integer. For example, in the given integer 8725, digits are
+   * [8, 7, 2, 5].
+   * @param number Any number of interest.
+   * @return The {@link IterableIntStream stream} of digits in the given integer.
+   */
+  public static IterableIntStream integerDigits(int number) {
+    var digitsInReverse = IntStream.iterate(
+      number,
+      n1 -> n1 != 0,
+      n1 -> n1 / 10
+    )
+      .map(n1 -> n1 % 10)
+      .toArray();
```

Comment:
We lose a lot of the benefit of a stream by going through an array, reversing it, and then creating another stream out of the array. I think it's actually ok to return the digits in reverse! This is what Ruby's [digits](https://ruby-doc.org/core-3.0.0/Integer.html#method-i-digits) method does.

For callers that need the digits in natural order, we can let them do the reversing themselves.

Reply:
I see. It makes sense. will do the changes.

Reply:
Addressed the comment